aws-distributed-configuration/src/main/java/io/micronaut/aws/distributedconfiguration/AwsDistributedConfigurationClient.java

```diff
@@ -73,38 +74,39 @@ public AwsDistributedConfigurationClient(AwsDistributedConfiguration awsDistribu
     @Override
     public Publisher<PropertySource> getPropertySources(Environment environment) {
         List<String> configurationResolutionPrefixes = generateConfigurationResolutionPrefixes(environment);
-
-        Map<String, Map> configurationResolutionPrefixesValues = new HashMap<>();
+        Map<String, Map<String, Map>> configurationResolutionPrefixKeyValueGroups = new LinkedHashMap<>();
+        int allKeysCount = 0;
 
         for (String prefix : configurationResolutionPrefixes) {
-            Optional<Map> keyValuesOptional = keyValueFetcher.keyValuesByPrefix(prefix);
-            if (keyValuesOptional.isPresent()) {
-                Map keyValues = keyValuesOptional.get();
-                configurationResolutionPrefixesValues.put(prefix, keyValues);
+            Optional<Map<String, Map>> keyValueGroupsOptional = keyValueFetcher.keyValuesByPrefix(prefix);
+            if (keyValueGroupsOptional.isPresent()) {
+                Map<String, Map> keyValueGroups = keyValueGroupsOptional.get();
+                configurationResolutionPrefixKeyValueGroups.put(prefix, keyValueGroups);
+                for (Map<String, ?> keyValues: keyValueGroups.values()) {
+                    allKeysCount += keyValues.size();
+                }
             }
         }
-        Set<String> allKeys = new HashSet<>();
-        for (Map m : configurationResolutionPrefixesValues.values()) {
-            allKeys.addAll(m.keySet());
-        }
-        Map<String, Object> result = new HashMap<>();
         if (LOG.isTraceEnabled()) {
-            LOG.trace("evaluating {} keys", allKeys.size());
+            LOG.trace("evaluating {} keys", allKeysCount);
         }
-        for (String k : allKeys) {
-            if (!result.containsKey(k)) {
-                for (String prefix : configurationResolutionPrefixes) {
-                    if (configurationResolutionPrefixesValues.containsKey(prefix)) {
-                        Map<String, ?> values = configurationResolutionPrefixesValues.get(prefix);
-                        if (values.containsKey(k)) {
-                            if (LOG.isTraceEnabled()) {
-                                LOG.trace("adding property {} from prefix {}", k, prefix);
-                            }
-                            result.put(k, values.get(k));
-                            break;
+        Map<String, Object> result = new HashMap<>();
```

Comment:
Use a linked hash map in case order is important

Reply:
There is a linked hash map that manages the order of precedence for all the active environments and their configuration resolutiion prefixes.